### ISSUE TITLE
Turn clickable table row into a hyperlink or a button

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#649]
***In GitLab by @lettfe on Jul 7, 2025, 08:53 GMT+2:***

# Refactoring Task

## :mount_fuji: Summary

Screenreaders need semantic information from the underlying html structure. If we want to enable people with screenreaders or keyboard users of the web browser to "click" anything it is not enough to provide the javascript to a normal html element like a div. Instead we need an HTML element that causes us to expect this behaviour, like a hyperlink or a button.

## :bomb: Why is the current state bad?

People with screenreaders cannot use the search for collections or containers because there is no semantic information about the table items being clickable.

## :goal: How could that be improved?

Turn the result list rows that can be clicked into hyperlinks or buttons while keeping the styling.

## :bulb: Hints

Do this in the DataTable component or even create a new ClickableDataTable component which will also refactor some of the code base since the clickable table rows are reimplemented in multiple components.

## :thinking: Risks & Open Questions

**Assignees:** lettfe

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/work_items/650*
